src/main/java/com/autosale/shop/security/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,47 @@
+package com.autosale.shop.security;
+
+import com.autosale.shop.service.JwtTokenService;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    private static final String JWT_TOKEN_PREFIX = "Bearer ";
+
+    private final JwtTokenService jwtTokenService;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+
+        String token = extractToken(request);
+        if (token != null) {
+            if (jwtTokenService.isValidToken(token)) {
+                Authentication authentication = jwtTokenService.getAuthentication(token);
+                SecurityContextHolder.getContext().setAuthentication(authentication);
+            }
+            filterChain.doFilter(request, response);
+        } else {
```

Comment:
remove else. filterChain.doFilter(request, response); can be in the end of the method one time